src/data/api-lapis.ts

```diff
@@ -276,6 +332,91 @@ function getInsertionEndpoint(sequenceType: SequenceType): string {
   }
 }
 
+// Helper function to convert URLSearchParams to filters object
+function urlSearchParamsToFilters(params: URLSearchParams): Record<string, any> {
+  const filters: Record<string, any> = {};
+  params.forEach((value, key) => {
+    // Skip accessKey as it goes in the URL, not the filters
+    if (key === 'accessKey') {
+      return;
+    }
+    // Try to parse as number if it looks like a number
+    const numValue = Number(value);
+    if (!isNaN(numValue) && value !== '') {
+      filters[key] = numValue;
+    } else {
+      filters[key] = value;
+    }
+  });
+  return filters;
+}
```

Comment:
This function makes it really easy for us to reuse all the code related to the URL params, but also maybe the number conversion could be problematic. I think we should be fine, but wanted to point it out anyways.

Reply:
Ah damn, I think maybe this is actually an issue for lists etc.